File: assets/samples/dns.lookup/apikey/ruby/dns_lookup_apikey.rb
require 'base64'
require 'openssl'
require 'open-uri'

domain = 'test.com'
key = 'your public dns lookup api key'
secret = 'your secret dns lookup api key'
username = 'your dns lookup api username'

time = (Time.now.to_f * 1000).to_i
req = Base64.encode64('{"t":' + time.to_s + ',"u":"' + username + '"}')
data = username + time.to_s + key
digest = OpenSSL::HMAC.hexdigest(OpenSSL::Digest::MD5.new, secret, data)

url = 'http://www.whoisxmlapi.com/whoisserver/DNSService?'\
    + 'requestObject=' + req + '&digest=' + digest + '&domainName=' + domain\
    + '&type=_all'

puts open(url).read

File: assets/samples/dns.lookup/apikey/nodejs/dns_lookup_apikey.js
var crypto = require('crypto');
var http = require('http');

var domain = 'example.com';
var key = 'your dns lookup api key';
var secret = 'your dns lookup api secret key';
var username = 'your dns lookup api username';

var time = (new Date).getTime();
var req = Buffer.from('{"t":'+time+',"u":"'+username+'"}').toString('base64');
var hmac = crypto.createHmac('md5', secret).update(username + time + key);
var digest = hmac.digest('hex');

var url = 'http://www.whoisxmlapi.com/whoisserver/DNSService?'
        + 'requestObject=' +req + '&digest=' +digest + '&domainName=' +domain
        + '&type=_all';

http.get(url, function(response) {
    var str = '';
    response.on('data', function(chunk) { str += chunk; });
    response.on('end', function() { console.log(str); });
}).end();